components/infobox/wikis/starcraft2/infobox_person_player.lua

```diff
@@ -0,0 +1,556 @@
+		military = string.lower(military)
+		local militaryCategory = ''
+		if String.Contains(military, 'starting') or String.Contains(military, 'pending') then
+			militaryCategory = '[[Category:Players waiting for Military Duty]]'
+			militaryStore = 'pending'
+		elseif String.Contains(military, 'ending') or String.Contains(military, 'started')
+			or String.Contains(military, 'ongoing') then
+				militaryCategory = '[[Category:Players on Military Duty]]'
+				militaryStore = 'ongoing'
+		elseif String.Contains(military, 'fulfilled') then
+			militaryCategory = '[[Category:Players expleted Military Duty]]'
+			militaryStore = 'fulfilled'
+		elseif String.Contains(military, 'exempted') then
+			militaryCategory = '[[Category:Players exempted from Military Duty]]'
+			militaryStore = 'exempted'
+		end
+
+		return display .. militaryCategory
+	end
+end
+
+function StarCraft2Player.getStatus(args)
+	if args.death_date then
+		statusStore = 'Deceased'
+	elseif args.retired then
+		statusStore = 'Retired'
+	elseif string.lower(args.role or 'player') ~= 'player' then
+		statusStore = 'not player'
+	end
+	return { store = statusStore }
+end
+
+function StarCraft2Player.addCustomCells(_, infobox, args)
+	local rank1, rank2
+	local yearsActive, activeCategory
+	if shouldStoreData and not statusStore then
+		rank1, rank2 = StarCraft2Player._getRank(pagename)
+		yearsActive, activeCategory = StarCraft2Player._get_matchup_data(pagename)
+		infobox:categories(activeCategory)
+	end
+
+	local currentYearEarnings = earningsGlobal[tostring(currentYear)]
+	if currentYearEarnings then
+		currentYearEarnings = '$' .. mw.language.new('en'):formatNum(currentYearEarnings)
+	end
+
+	infobox	:cell('Approx. Earnings ' .. currentYear, currentYearEarnings)
+			:cell(rank1.name or 'Rank', rank1.rank)
+			:cell(rank2.name or 'Rank', rank2.rank)
+			:cell('Military Service', StarCraft2Player._military(args.military))
+			:cell('Years active', yearsActive)
+
+	return infobox
+end
+
+function StarCraft2Player._get_matchup_data(player)
+	local count
+	local category = ''
+	player = string.gsub(player, '_', ' ')
+	local conditions = '[[opponent::' .. player .. ']] AND [[walkover::]] AND [[winner::>]]'
+
+	local data = {} -- get LPDB results in here
+	local offset = 0
+	repeat
+		local additional_data = mw.ext.LiquipediaDB.lpdb('match2', {
+			conditions = conditions,
+			query = 'match2opponents, date',
+			offset = offset,
+			limit = 5000
+		})
+		count = 0
+		-- Merging
+		for i, item in ipairs(additional_data) do
+			data[offset + i] = item
+			count = count + 1
+		end
+		offset = offset + count
+	until count ~= 5000
+
+	local yearsActive = ''
+	local years = {}
+	local vs = {}
+	local races = { 'p', 't', 'z', 'r', 'total' }
+	for _, item1 in pairs(races) do
+		vs[item1] = {}
+		for _, item2 in pairs(races) do
+			vs[item1][item2] = { ['win'] = 0, ['loss'] = 0 }
+		end
+	end
+
+	if type(data[1]) == 'table' then
+		local CleanRace = { ['t'] = 't', ['z'] = 'z', ['p'] = 'p'}
```

Comment:
Const at the top?
Seems like something that's used in many modules as well? Move the cleaning to a standalone shared module?